api/python/quilt3/backends/__init__.py

```diff
@@ -0,0 +1,19 @@
+from quilt3.util import PhysicalKey, get_from_config, fix_url
+
+from .base import PackageRegistry
+from .local import LocalPackageRegistryV1
+from .s3 import S3PackageRegistryV1
+
+
+def get_package_registry(path=None) -> PackageRegistry:
```

Comment:
Let's make have a signature that's consistent with `PhysicalKey.from_path`. Users should also have access to PhysicalKey since that class is part of the API (e.g., `Package.resolve_hash`).

Reply:
`Package.resolve_hash()` was added in https://github.com/quiltdata/quilt/commit/8efb0b6a9bac8108b9fde542c0465d487ae6f01f and `registry` parameter was `str`. `PhysicalKey` was added in https://github.com/quiltdata/quilt/pull/1478 and was intended to be private. This PR also made `Package.shorten_tophash()` [private](https://github.com/quiltdata/quilt/pull/1478#discussion_r362641345). So `Package.resolve_hash()` worked for users only in 3.1.7 and 3.1.8. I think we should either make it private or make it take `str` again. Make it private would be easier.

Reply:
I don't mind making it private. It does seem useful to be able to read a package from a short hash though.